File: FileManagerDemo/FileManagerDemo/ViewController.m
//
//  ViewController.m
//  FileManagerDemo
//
//  Created by kuroky on 2017/6/23.
//  Copyright © 2017年 kuroky. All rights reserved.
//

#import "ViewController.h"
#import <MXFileManager-Swift.h>

@interface ViewController () <UITableViewDataSource, UITableViewDelegate>

@property (nonatomic, strong) NSArray *dataList;
@property (strong, nonatomic) IBOutlet UITableView *tableView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setupData];
    [self setupUI];
}

- (void)setupData {
    self.dataList = @[@"创建文件夹", @"创建临时文件夹", @"创建文件", @"创建临时文件", @"自动清除缓存", @"手动清除缓存", @"遍历文件tmp", @"遍历文件cache", @"文件大小"];
}

- (void)setupUI {
    self.tableView.rowHeight = 49;
    self.tableView.separatorInset = UIEdgeInsetsZero;
    [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:@"cellId"];
}

#pragma mark - UITableViewDataSource
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.dataList.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cellId"];
    cell.textLabel.text = self.dataList[indexPath.row];
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
    NSString *title = self.dataList[indexPath.row];
    if ([title isEqualToString:@"创建文件夹"]) {
        static BOOL storage = YES;
        NSString *name = [@"cache" stringByAppendingString:[NSUUID UUID].UUIDString];
        [[MXFileManager fileManager] createDirectioryWithDirName:name
                                                           isTmp:NO
                                                   shouldStorage:storage
                                                      completion:^(NSString *filePath) {
                                                          NSLog(@"创建文件夹: %@", filePath);
                                                      }];
        storage = !storage;
    }
    else if ([title isEqualToString:@"创建临时文件夹"]) {
        static BOOL storage = YES;
        NSString *name = [@"tmp" stringByAppendingString:[NSUUID UUID].UUIDString];
        [[MXFileManager fileManager] createDirectioryWithDirName:name
                                                           isTmp:YES
                                                   shouldStorage:storage
                                                      completion:^(NSString *filePath) {
                                                          NSLog(@"创建临时文件夹: %@", filePath);
                                                      }];
        storage = !storage;
    }
    else if ([title isEqualToString:@"创建文件"]) {
        static BOOL storage = YES;
        NSString *name = [NSString stringWithFormat:@"cache_%@.txt", [NSUUID UUID].UUIDString];
        [[MXFileManager fileManager] createFileWithName:name
                                                  isTmp:NO
                                          shouldStorage:storage
                                             completion:^(NSString *filePath) {
                                                 NSLog(@"创建文件: %@", filePath);
                                             }];
        storage = !storage;
    }
    else if ([title isEqualToString:@"创建临时文件"]) {
        static BOOL storage = YES;
        NSString *name = [NSString stringWithFormat:@"tmp_%@.txt", [NSUUID UUID].UUIDString];
        [[MXFileManager fileManager] createFileWithName:name
                                                  isTmp:YES
                                          shouldStorage:storage
                                             completion:^(NSString *filePath) {
                                                 NSLog(@"创建临时文件: %@", filePath);
                                             }];
        storage = !storage;
    }
    else if ([title isEqualToString:@"自动清除缓存"]) {
        [[MXFileManager fileManager] clearTmpDataWithCompletion:^{
            NSLog(@"自动清除缓存");
        }];
    }
    else if ([title isEqualToString:@"手动清除缓存"]) {
        [[MXFileManager fileManager] clearCacheDataWithCompletion:^{
            NSLog(@"手动清除缓存");
        }];
    }
    else if ([title isEqualToString:@"遍历文件tmp"]) {
        
        [[MXFileManager fileManager] enumeratorForTmpWithTmp:YES
                                                  completion:^(NSArray<NSString *> *files) {
                                                      NSLog(@"files : %@", files);
                                                  }];
    }
    else if ([title isEqualToString:@"遍历文件cache"]) {
        [[MXFileManager fileManager] enumeratorForTmpWithTmp:NO completion:^(NSArray<NSString *> *files) {
            NSLog(@"files : %@", files);
        }];
    }
    else if ([title isEqualToString:@"文件大小"]) {
        [[MXFileManager fileManager] getSizeWithCompletion:^(double size) {
            NSLog(@"size: %lu", (unsigned long)size);
        }];
        
    }
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: MXFileManageriOS/MXFileManageriOS/ViewController.swift
//
//  ViewController.swift
//  MXFileManageriOS
//
//  Created by kuroky on 2019/6/12.
//  Copyright © 2019 Emucoo. All rights reserved.
//

import UIKit

import MXFileManager

class ViewController: UIViewController {

    @IBOutlet weak var tableView: UITableView!
    let dataList = ["cache文件夹", "文件", "临时文件夹", "所占文件大小", "清除tmp", "清除cache路径"]
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.tableView.rowHeight = 50
        self.tableView.register(UITableViewCell.self, forCellReuseIdentifier: "cell")
    }
}

extension ViewController: UITableViewDataSource, UITableViewDelegate {
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.dataList.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath)
        let text = dataList[indexPath.row]
        cell.textLabel?.text = text
        return cell
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        
        let text = dataList[indexPath.row]
        if text == "cache文件夹" {
            MXFileManager.fileManager.createDirectiory(dirName: "1123") { path in
                if let path = path {
                    print(path)
                }
            }
        }
        else if text == "文件" {
            MXFileManager.fileManager.createFile(name: "456") { path in
                if let path = path {
                    print(path)
                }
            }
        }
        else if text == "所占文件大小" {
            MXFileManager.fileManager.getSize { size in
                print(size)
            }
        }
        else if text == "清除tmp" {
            MXFileManager.fileManager.clearTmpData {
                print("tmp clear")
            }
        }
        else if text == "清除cache路径" {
            MXFileManager.fileManager.clearCacheData {
                print("cache clear")
            }
        }
    }
}

